types/test.tsx

```diff
@@ -35,13 +36,19 @@ export async function testPureRender() {
 
   // helpers
   const {container, rerender, debug} = page
+  expectType<HTMLElement, typeof container>(container)
   return {container, rerender, debug}
 }
 
 export function testRenderOptions() {
   const container = document.createElement('div')
   const options = {container}
-  render(<div />, options)
+  const {container: returnedContainer} = render(<button />, options)
+  // Unclear why TypeScript infers `HTMLElement` here when the the input `container` is `HTMLDivElement`.
```

Comment:
If we'd swap the [`render` methods in the declaration file](https://github.com/testing-library/react-testing-library/blob/b9836da4d4d2c26494f53b3b6cb629799774615b/types/index.d.ts#L49-L59), this would infer correctly.
I'm unfamiliar with the types for RTL, so I'm not sure if this would break something, or if we could maybe entirely remove the other `render` type.

```suggestion
  // Unclear why TypeScript infers `HTMLElement` here when the input `container` is `HTMLDivElement`.
```

Reply:
@timdeschryver Awesome, thanks! Fixed in [`a61a0d5` (#868)](https://github.com/testing-library/react-testing-library/pull/868/commits/a61a0d5f7da0acdb72312281d3d02e225e30d288)

Reply:
I'm glad I could help 🙂